sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -7294,11 +7294,6 @@ pi_result piextUSMEnqueuePrefetch(pi_queue Queue, const void *Ptr, size_t Size,
   // Lock automatically releases when this goes out of scope.
   std::lock_guard<std::mutex> lock(Queue->PiQueueMutex);
 
-  _pi_ze_event_list_t TmpWaitList;
-  if (auto Res = TmpWaitList.createAndRetainPiZeEventList(NumEventsInWaitList,
-                                                          EventWaitList, Queue))
-    return Res;
-
   // Get a new command list to be used on this call
```

Comment:
We already call createAndRetainPiZeEventList for the EventWaitList at line 7313 below.
This code unnecessarily increment the reference count twice, which leads to memory leak.

Reply:
In all other cases we create `TmpWaitList` before creating the real Event (and I don't remember why now). Please investigate/document and make your change in line with the rest of the code (presumably remove the other call to createAndRetainPiZeEventList, not this one)